src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -167,9 +152,77 @@ public void Delete(string key, CookieOptions options)
         });
     }
 
+    private bool TryGetLogger([NotNullWhen(true)] out ILogger? logger)
+    {
+        if (!_retrievedLogger)
+        {
+            _retrievedLogger = true;
+            var services = _features.Get<IServiceProvidersFeature>()?.RequestServices;
+            _logger = services?.GetService<ILogger<ResponseCookies>>();
+        }
+
+        logger = _logger;
+        return logger is not null;
+    }
+
+    private static MessagesToLog GetMessagesToLog(CookieOptions options)
+    {
+        var toLog = MessagesToLog.None;
+
+        if (!options.Secure && options.SameSite == SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.SameSiteNotSecure;
+        }
+
+        if (!options.Secure && options.Partitioned)
+        {
+            toLog |= MessagesToLog.PartitionedNotSecure;
+        }
+
+        if (options.Partitioned && options.SameSite != SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.PartitionedNotSameSiteNone;
+        }
+
+        return toLog;
+    }
+
+    private static void LogMessages(ILogger logger, MessagesToLog messages, string cookieName)
+    {
+        if ((messages & MessagesToLog.SameSiteNotSecure) != 0)
+        {
+            Log.SameSiteCookieNotSecure(logger, cookieName);
+        }
+
+        if ((messages & MessagesToLog.PartitionedNotSecure) != 0)
+        {
+            Log.PartitionedCookieNotSecure(logger, cookieName);
+        }
+
+        if ((messages & MessagesToLog.PartitionedNotSameSiteNone) != 0)
+        {
+            Log.PartitionedCookieNotSameSiteNone(logger, cookieName);
```

Comment:
Wouldn't be a lot simpler to make this non-static? For example:

```csharp
    private void LogMessagesForBadOptions(CookieOptions options)
    {
        if (!TryGetLogger(out var logger))
        {
            return;
        }

        if (!options.Secure && options.SameSite == SameSiteMode.None)
        {
            logger.SameSiteCookieNotSecure(logger, cookieName);
        }

        // ....
```

I don't see what the flags enum is buying us other than a bunch of extra lines of code.

Reply:
I was trying to (a) eliminate the duplicated checking code and (b) avoiding repeating the checks for each cookie in the list overload.  I don't like the flags enum either, but it seemed relatively future-proof and more performant than some sort of lambda.

I find the static method more readable, but I don't feel strongly about it.